### PR TITLE
MovementSensor

### PR DESCRIPTION
**Pull request by zaporter-work**
Howdy!

I needed MovementSensor bindings for a personal project so I created them. 

I don't have time to polish this up and bring it through a PR, but I thought I would share the code in case someone wanted to take it from here. If everyone is busy, that is fine too! Just thought I should share.

There are a few things to figure out (and the linter is complaining about a few missing `const`s), but the general structure is there and the tests pass on my machine...

Weirdest part of this PR is that the file is named `movementsensor/movementsensor.hpp`.

Pros of `movementsensor.hpp`:
- rdk has `movementsensor.go`
- api has `movementsensor.proto`
- protos are generated as `movementsensor`, not `movement_sensor`

Cons of `movementsensor.hpp`:
- `movement_sensor.hpp` would be easier to read and would make more sense
- Python calls the file `movement_sensor.py`
- Ts calls the file `movement-sensor.ts`

### CODE REVIEW

File: src/viam/sdk/components/movementsensor/movementsensor.hpp [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        double value;
+    };
+
+    // @struct orientation
+    // @brief MovementSensor is the only user of viam::common::v1::Orientation right now.
+    // Unfortunately, that is not the same as viam::app::v1::Orientation (which is what spatialmath
+    // uses).
+    struct orientation {
+        double o_x, o_y, o_z, theta;
+        // TODO(pre-merge) move the to/from protos in here. Nicer that way
```
Comment (zaporter-work) [inline src/viam/sdk/components/movementsensor/movementsensor.hpp:58]: Maybe? I saw you do this for some other methods? I kept the current way with everything in the component..

Not sure what the plan is

File: src/viam/sdk/components/movementsensor/server.cpp [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    std::unordered_map<std::string, float> result = movementsensor->get_accuracy(extra);
+    response->mutable_accuracy()->empty();
+    for (const auto& i : result) {
+        response->mutable_accuracy()->emplace(i.first, i.second);
```
Comment (zaporter-work) [inline src/viam/sdk/components/movementsensor/server.cpp:216]: Strangely this like works on my laptop but does not in CI. Different proto versions maybe?
``` 
/tmp/src/viam/sdk/components/movementsensor/server.cpp:216:39: error: no member named 'emplace' in 'google::protobuf::Map<std::basic_string<char>, float>' [clang-diagnostic-error]
        response->mutable_accuracy()->emplace(i.first, i.second);
```

Comment (acmorrow) [reply src/viam/sdk/components/movementsensor/server.cpp:216]: Yes, almost certainly different proto version. I remember encountering this same issue when working on the MLModel stuff and had to stop using `emplace`.

**Comment by zaporter-work** [comment]
Implemented in https://github.com/viamrobotics/viam-cpp-sdk/pull/171

**Pull request closed without merging**